### PR TITLE
Implement PublishAsAzureRedis

### PR DESCRIPTION
**Pull request by davidfowl**
- Added support for publishing a redis resource as an azure redis service.
- Added ConnectionStringExpression and ConnectionStringReferenceExpression to IResourceWithConnectionString.

Contributes to #1960 
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2188)

### CODE REVIEW

File: src/Aspire.Hosting.Azure/Bicep/AzureBicepAppConfigurationResource.cs [hunk 1]
```diff
@@ -13,6 +13,11 @@ public class AzureBicepAppConfigurationResource(string name) :
     AzureBicepResource(name, templateResouceName: "Aspire.Hosting.Azure.Bicep.appconfig.bicep"),
     IResourceWithConnectionString
 {
+    /// <summary>
```
Comment (mitchdenny) [inline src/Aspire.Hosting.Azure/Bicep/AzureBicepAppConfigurationResource.cs:16]: XML docs in the wrong place?

File: src/Aspire.Hosting.Azure/Bicep/AzureBicepPostgresResource.cs [hunk 2]
```diff
@@ -12,19 +12,25 @@ namespace Aspire.Hosting.Azure;
...
     AzureBicepResource(name, templateResouceName: "Aspire.Hosting.Azure.Bicep.postgres.bicep"),
     IResourceWithConnectionString
 {
     internal List<string> Databases { get; } = [];
 
+    /// <summary>
+    /// Gets the connection template for the manifest for the Azure Postgres Flexible Server.
+    /// </summary>
+    public string ConnectionStringExpression =>
+        $"Host={{{Name}.outputs.pgfqdn}};Username={username};Password={password.ValueExpression}";
```
Comment (mitchdenny) [inline src/Aspire.Hosting.Azure/Bicep/AzureBicepPostgresResource.cs:25]: I'm not sure I like these expression strings being on the public API surface. Seems like an internal implementation detail.

File: src/Aspire.Hosting.Azure/Bicep/AzureBicepAppConfigurationResource.cs (hunk 1 above)
Comment (davidfowl) [reply src/Aspire.Hosting.Azure/Bicep/AzureBicepAppConfigurationResource.cs:16]: What do you mean?

File: src/Aspire.Hosting.Azure/Bicep/AzureBicepPostgresResource.cs (hunk 2 above)
Comment (davidfowl) [reply src/Aspire.Hosting.Azure/Bicep/AzureBicepPostgresResource.cs:25]: Why is GetConnectionString public but this isn't?

**Review by mitchdenny — APPROVED** [review]

**Pull request merged**